### PR TITLE
test: Add tests for CNS IPAM invoker methods

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: cni/network/network_test.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func TestPlugin(t *testing.T) {
 		Options: make(map[string]interface{}),
 	}
 	plugin.nm.CreateNetwork(nwInfo)
-	//	plugin.Delete(args)
+	// plugin.Delete(args)
```
[the remark this answers is not in the document]
Comment (matmerr) [reply cni/network/network_test.go:77]: but this was just a formatting change, we can remove it when your changes to add are included

File: cni/network/network.go [hunk 2]
```diff
@@ -11,14 +11,15 @@ import (
...
+	"strconv"
 	"strings"
 	"time"
 
 	"github.com/Azure/azure-container-networking/aitelemetry"
 	"github.com/Azure/azure-container-networking/cni"
 	"github.com/Azure/azure-container-networking/cni/api"
 	"github.com/Azure/azure-container-networking/cns"
-	cnsclient "github.com/Azure/azure-container-networking/cns/client"
+	cnsc "github.com/Azure/azure-container-networking/cns/client"
```
Comment (matmerr) [inline cni/network/network.go:22]: import name subject to opinions, no preference here, avoiding collision

Comment (tamilmani1989) [reply cni/network/network.go:22]: can be cnscli?

Comment (rbtr) [reply cni/network/network.go:22]: cnscli is what i have called the client elsewhere to keep it from conflicting with the cnsclient import

File: cni/network/invoker_cns.go [hunk 3]
```diff
@@ -38,27 +42,34 @@ type IPv4ResultInfo struct {
...
+	args *cniSkel.CmdArgs,
+	hostSubnetPrefix *net.IPNet,
+	options map[string]interface{}) (*cniTypesCurr.Result, *cniTypesCurr.Result, error) {
 	// Parse Pod arguments.
 	podInfo := cns.KubernetesPodInfo{
 		PodName:      invoker.podName,
 		PodNamespace: invoker.podNamespace,
 	}
+
+	log.Printf(podInfo.PodName)
```
Comment (tamilmani1989) [inline cni/network/invoker_cns.go:65]: why just printing podname alone here?

File: cni/network/network.go [hunk 4]
```diff
@@ -740,15 +743,15 @@ func (plugin *NetPlugin) createEndpointInternal(
 
 	setEndpointOptions(cnsNetworkConfig, &epInfo, vethName)
 
-	cnscli, err := cnsclient.New(nwCfg.CNSUrl, defaultRequestTimeout)
+	cnsclient, err := cnscli.New(nwCfg.CNSUrl, defaultRequestTimeout)
```
Comment (tamilmani1989) [inline cni/network/network.go:746]: cnsclient intialized multiple times with different args

File: cni/network/network.go [hunk 5]
```diff
@@ -917,24 +920,26 @@ func (plugin *NetPlugin) Delete(args *cniSkel.CmdArgs) error {
...
 		switch nwCfg.Ipam.Type {
 		case network.AzureCNS:
-			plugin.ipamInvoker, err = NewCNSInvoker(k8sPodName, k8sNamespace)
+			cnsURL := "http://localhost:" + strconv.Itoa(cnsPort)
+			cnsClient, er := cnscli.New(cnsURL, defaultRequestTimeout)
 			if err != nil {
-				log.Printf("[cni-net] Creating network %v failed with err %v.", networkId, err)
-				return err
+				log.Printf("[cni-net] failed to create cns client", networkId, err)
+				return fmt.Errorf("ailed to create cns client with err %w", er)
```
Comment (tamilmani1989) [inline cni/network/network.go:927]: typo : "failed"

**Review by tamilmani1989 — APPROVED** [review]
approving. discussed offline..comments will be addressed in another PR

File: cni/network/network.go (hunk 4 above)
Comment (tamilmani1989) [reply cni/network/network.go:746]: @rbtr need to sync with you on this

**Pull request merged**